sct-commons/src/main/java/org/lfenergy/compas/sct/commons/LNodeStatusService.java

```diff
@@ -0,0 +1,100 @@
+// SPDX-FileCopyrightText: 2024 RTE FRANCE
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package org.lfenergy.compas.sct.commons;
+
+import lombok.RequiredArgsConstructor;
+import org.lfenergy.compas.scl2007b4.model.*;
+import org.lfenergy.compas.sct.commons.domain.DataAttribute;
+import org.lfenergy.compas.sct.commons.domain.DoLinkedToDa;
+import org.lfenergy.compas.sct.commons.domain.DoLinkedToDaFilter;
+import org.lfenergy.compas.sct.commons.dto.SclReportItem;
+import org.lfenergy.compas.sct.commons.util.CommonConstants;
+import org.lfenergy.compas.sct.commons.util.PrivateUtils;
+import org.lfenergy.compas.sct.commons.util.SclConstructorHelper;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+@RequiredArgsConstructor
+public class LNodeStatusService {
+
+    private static final String LNODE_STATUS_PRIVATE_TYPE = "COMPAS-LNodeStatus";
+    private final LdeviceService ldeviceService;
+    private final LnService lnService;
+    private final DataTypeTemplatesService dataTypeTemplatesService;
+
+    public List<SclReportItem> updateLnModStValBasedOnLNodeStatus(SCL scl) {
+        return scl.getSubstation().stream()
+                .flatMap(tSubstation -> tSubstation.getVoltageLevel().stream())
+                .flatMap(tVoltageLevel -> tVoltageLevel.getBay().stream())
+                .flatMap(tBay -> tBay.getFunction().stream())
+                .flatMap(tFunction -> tFunction.getLNode().stream())
+                .map(tlNode -> updateSingleLnModStValBasedOnLNodeStatus(scl, tlNode))
+                .filter(Objects::nonNull)
+                .toList();
+    }
+
+    private SclReportItem updateSingleLnModStValBasedOnLNodeStatus(SCL scl, TLNode tlNode) {
+        String lNodeLNS = PrivateUtils.extractStringPrivate(tlNode, LNODE_STATUS_PRIVATE_TYPE).orElse(null);
```

Comment:
Isn't namespaced private ?

```xml
 <Private type="COMPAS-LNodeStatus">
       <compas:LNodeStatus>off</compas:LNodeStatus>
 </Private>
```
Instead of:
```xml
 <Private type="COMPAS-LNodeStatus">
       off
 </Private>
```

```suggestion
        String lNodeLNS = PrivateUtils.extractCompasPrivate(tlNode, TCompasLNodeStatus.class).map(TCompasLNodeStatus::value).orElse(null);
```

Reply:
It really is a String directly in the content of the Private element, like that :
 `<Private type="COMPAS-LNodeStatus">on</Private>`.

It is like that in the SSD files provided by the PO.
The private is also created by RCONF like that when converting ExportSite to SSD.

tCompasLNodeStatus was added by mistake in the SCL_CoMPAS.xsd. This type is never used.

Reply:
OK